### DISCUSSION TITLE
Why does my repo produce errant output from ch.qos.logback ?

### DISCUSSION THREAD

**Discussion by Jonathan-Zollinger** [micronaut-projects/micronaut-core#10926] [General] [answered]
I'm getting weird logging prior to my program's execution. What have I set up wrong? I've tried facilitating this explanation with pertinent screenshots. I've collapsed them to try and keep this readabale as well.

## Summary
<details><summary>My CLI produces two log outputs, one which I can change by setting the log level in <code>logback.xml</code>, but the logs from <code>ch.qos.logback</code> I can't get rid of.</summary>

![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/887f8659-79db-4518-bebe-917943e0ad7f)

</details>

## Troubleshooting
<details><summary>Changing log levels from info to ERROR, removes the info log output with color(as expected), but the info logs from <code>ch.qos.logback</code> remain
</summary>

![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/29fb670d-8e5d-4c69-a7ce-902cd0881bc2)

```shell
 target/threshr-cli -h
10:54:31,055 |-INFO in ch.qos.logback.classic.LoggerContext[default] - This is logback-classic version 1.5.3
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - No custom configurators were discovered as a service.
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - Trying to configure with ch.qos.logback.classic.joran.SerializedModelConfigurator
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - Constructed configurator of type class ch.qos.logback.classic.joran.SerializedModelConfigurator
10:54:31,055 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.scmo]
10:54:31,055 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.scmo]
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - ch.qos.logback.classic.joran.SerializedModelConfigurator.configure() call lasted 0 milliseconds. ExecutionStatus=INVOKE_NEXT_IF_ANY
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - Trying to configure with ch.qos.logback.classic.util.DefaultJoranConfigurator
10:54:31,055 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - Constructed configurator of type class ch.qos.logback.classic.util.DefaultJoranConfigurator
10:54:31,055 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
10:54:31,055 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [resource:/logback.xml]
10:54:31,055 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@8a3b282 - Resource [logback.xml] occurs multiple times on the classpath.
10:54:31,055 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@8a3b282 - Resource [logback.xml] occurs at [resource:/logback.xml#1]
10:54:31,055 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@8a3b282 - Resource [logback.xml] occurs at [resource:/logback.xml]
10:54:31,055 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@2110bb19 - URL [resource:/logback.xml] is not of type file
10:54:31,056 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - Processing appender named [STDOUT]
10:54:31,056 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
10:54:31,057 |-INFO in ch.qos.logback.core.model.processor.ImplicitModelHandler - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
10:54:31,057 |-INFO in ch.qos.logback.classic.model.processor.RootLoggerModelHandler - Setting level of ROOT logger to ERROR
10:54:31,057 |-INFO in ch.qos.logback.core.model.processor.AppenderRefModelHandler - Attaching appender named [STDOUT] to Logger[ROOT]
10:54:31,057 |-INFO in ch.qos.logback.core.model.processor.DefaultProcessor@198c6120 - End of configuration.
10:54:31,057 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@189f6866 - Registering current configuration as safe fallback point
10:54:31,057 |-INFO in ch.qos.logback.classic.util.ContextInitializer@1e911e66 - ch.qos.logback.classic.util.DefaultJoranConfigurator.configure() call lasted 2 milliseconds. ExecutionStatus=DO_NOT_INVOKE_NEXT_IF_ANY

Usage: pdp [-hV] -s=<store> -t=<tcinValues>
  -h, --help                Show this help message and exit.
  -s, --store-id=<store>    store id as given in redsky api
  -t, --tcin=<tcinValues>
  -V, --version             Print version information and exit.
```

</details>

<details><summary>Compared to <a href="https://guides.micronaut.io/latest/micronaut-cli-jwkgen-maven-java.html">jwkgen guide</a> which I've produced locally, the log output is only the expected info log.</summary>

![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/d026091a-a57c-464d-b414-6145f660a804)
![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/8acb6127-f069-4a1e-83a3-099499bfd69e)

</details>

<details><summary>This makes me think I've configured something wrong with my logging dependencies, but the two repos use the same logback dependency and scope, they both use micronaut 4.5.0 as well.</summary>

### threshr dependency
![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/9a7cf031-1d27-44bf-86af-a8d9691efc3e)

### jwk dependency
![image](https://github.com/micronaut-projects/micronaut-core/assets/62955101/0ddf82b4-1364-4d50-9d32-1a835b4699f2)

</details>

any ideas why I'm getting such different output? this is from my [threshr-cli repo](https://github.com/Graqr/threshr-cli/tree/introduce-pdp-command). [download threshr-cli-0.0.0.zip](https://github.com/user-attachments/files/15931053/threshr-cli-0.0.0.zip)

<details><summary>My environment</summary>
OS: Fedora 40 workstation

Micronaut Version: 4.5.0
JVM: graal-ce 17.0.9

`./mvnw --version` output:
```shell
Java version: 17.0.9, vendor: GraalVM Community, runtime: /home/jimmy/.sdkman/candidates/java/17.0.9-graalce
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.9.4-200.fc40.x86_64", arch: "amd64", family: "unix"
```
</details>

**Comment by Jonathan-Zollinger** [comment]
fwiw, following similar [thread discussions](https://stackoverflow.com/questions/3401051/suppress-all-logback-output-to-console) and manually setting the `ch.qos.logback` library to off in the logback config doesn't seem to produce anything different. 

```xml
<configuration>

    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <!-- encoders are assigned the type
             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
        <encoder>
            <pattern>%cyan(%d{HH:mm:ss.SSS}) %gray([%thread]) %highlight(%-5level) %magenta(%logger{36}) - %msg%n</pattern>
        </encoder>
    </appender>

    <root level="ERROR">
        <appender-ref ref="STDOUT" />
    </root>
    <logger name="ch.qos.logback" level="OFF" additivity="false" />
</configuration>
```

**Comment by yawkat** [comment]
I think the problem is those WARN entries in the logback status log (duplicate logback.xml). If you fix those, it should not print the INFO status log either.

**Reply by Jonathan-Zollinger** [reply]
I'm sorry, I'm not following - I see only one logback.xml in my repo at my src/main/resources directory.

**Comment by BramMeerten** [comment, marked as answer]
I had the same issue, solved it by adding the following line to my logback.xml:
```xml
<statusListener class="ch.qos.logback.core.status.NopStatusListener" />
```

Got the answer from [this stackoverflow post](https://stackoverflow.com/questions/65035615/how-to-disable-ch-qos-logbacks-own-info-messages-in-the-run-of-the-springbootte).

**Reply by Jonathan-Zollinger** [reply]
sweet, thanks @BramMeerten !
